ui/app/AppLayouts/Wallet/popups/swap/SwapModal.qml

```diff
@@ -52,91 +64,103 @@ StatusDialog {
+        ColumnLayout {
+            Layout.fillWidth: true
+            spacing: 0
+            StatusBaseText {
+                text: "This area is a temporary placeholder"
+                font.bold: true
             }
-        }
-        StatusButton {
-            text: "Send Approve Tx"
-            onClicked: {
-                swapAdaptor.sendApproveTx()
+            /* TODO: Will be swapped out under https://github.com/status-im/status-desktop/issues/14825
+               Will also handle that if one input is being edited the other one should be in loading state in that task */
+            StatusBaseText {
+                text: qsTr("Selected from token: %1").arg(swapInputParamsForm.fromTokensKey)
             }
-        }
-        StatusButton {
-            text: "Send Swap Tx"
-            onClicked: {
-                swapAdaptor.sendSwapTx()
+            StatusInput {
+                id: fromTokenAmountInput
+                Layout.fillWidth: true
+                text: swapInputParamsForm.fromTokenAmount
+                onTextChanged: {
+                    swapInputParamsForm.fromTokenAmount = text
+                    root.swapInputParamsForm.lastEditedAmount = SwapInputParamsForm.LastEditedAmount.FromToken
+                }
             }
-        }
-        StatusScrollView {
-            Layout.fillWidth: true
-            Layout.preferredHeight: 200
-
-            StatusTextArea {
-                text: {
-                    let routes = SQUtils.ModelUtils.modelToArray(swapAdaptor.suggestedRoutes)
-                    let routesString = JSON.stringify(routes, null, "  ")
-                    return qsTr("Suggested routes: \n%1").arg(routesString)
+            /* TODO: Will be swapped out under https://github.com/status-im/status-desktop/issues/14826
+               Will also handle that if one input is being edited the other one should be in loading state in that task */
+            StatusBaseText {
+                text: qsTr("Selected to token: %1").arg(swapInputParamsForm.toTokenKey)
+            }
+            StatusInput {
+                id: toTokenAmountInput
+                Layout.fillWidth: true
+                text: root.swapAdaptor.validSwapProposalReceived && root.swapAdaptor.toToken ?
+                          root.swapAdaptor.formatCurrencyAmount(root.swapAdaptor.swapOutputData.toTokenAmount,
+                                                                root.swapAdaptor.toToken.symbol,
+                                                                {"minDecimals": root.swapAdaptor.toToken.decimals,
+                                                                    "stripTrailingZeroes": true, "noSymbol": true}) :
+                          root.swapInputParamsForm.toTokenAmount
+                onTextChanged: {
+                    root.swapInputParamsForm.lastEditedAmount = SwapInputParamsForm.LastEditedAmount.ToToken
+                    if (!root.swapAdaptor.validSwapProposalReceived) {
```

Comment:
is this is supposed to catch the case where the text is changed due to an incoming swap proposal?

Reply:
handled here https://github.com/status-im/status-desktop/pull/15161/files#:~:text=tokenAmount%3A%20root.swapAdaptor.validSwapProposalReceived%20%26%26%20root.swapAdaptor.toToken%20%3F%20root.swapAdaptor.swapOutputData.toTokenAmount%3A%20root.swapInputParamsForm.toTokenAmount